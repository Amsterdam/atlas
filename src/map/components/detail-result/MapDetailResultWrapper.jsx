import React from 'react';
import PropTypes from 'prop-types';

import MaximizeIcon from '../../../../public/images/icon-maximize.svg';

const MapDetailResultWrapper = ({ children, panoUrl, subTitle, title, onMaximize }) => (
  <section className="map-detail-result">
    <header
      className={`
        map-detail-result__header
        map-detail-result__header--${panoUrl ? 'pano' : 'no-pano'}
      `}
    >
      {panoUrl && (
        <img
          alt="Panoramabeeld"
          className="map-detail-result__header-pano"
          src={panoUrl}
        />
      )}
      <div className="map-detail-result__header-container">
        <h1 className="map-detail-result__header-title">{title}</h1>
        {subTitle && (
          <h2 className="map-detail-result__header-subtitle">{subTitle}</h2>
        )}
      </div>
    </header>
    {children && (
      <div className="map-detail-result__scroll-wrapper">
        {children}
        <footer className="map-search-results__footer">
          <button onClick={onMaximize} className="map-search-results__button">
            <MaximizeIcon className="map-search-results__button-icon" />
            Volledig weergeven
          </button>
        </footer>
      </div>
    )}
  </section>
);

MapDetailResultWrapper.defaultProps = {
  children: null,
  subTitle: ''
};

MapDetailResultWrapper.propTypes = {
  children: PropTypes.element,
  panoUrl: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  title: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired
};

export default MapDetailResultWrapper;
