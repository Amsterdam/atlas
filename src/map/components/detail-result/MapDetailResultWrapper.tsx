import { Heading, Button, Link, themeSpacing, themeColor } from '@amsterdam/asc-ui'
import { LatLngLiteral } from 'leaflet'
import { FunctionComponent } from 'react'
import { useSelector } from 'react-redux'
import styled from 'styled-components'
import PanoAlert from '../../../app/components/PanoAlert/PanoAlert'
import useGetLegacyPanoramaPreview from '../../../app/utils/useGetLegacyPanoramaPreview'
import Maximize from '../../../shared/assets/icons/icon-maximize.svg'
import { isEmbedded } from '../../../shared/ducks/ui/ui'
import { getUser } from '../../../shared/ducks/user/user'
import useCheckPanoramaPermission from '../../../app/utils/useCheckPanoramaPermission'

export interface MapDetailResultWrapperProps {
  title: string
  location?: LatLngLiteral
  subTitle?: string | null
  onMaximize: () => void
  children: React.ReactNode
}

const StyledLink = styled(Link)`
  padding: 0;
  height: 100%;
  width: 100%;
`

const Header = styled.header`
  margin: 0 ${themeSpacing(3)};
`

const StyledButton = styled(Button)`
  background-color: ${themeColor('tint', 'level1')};
  svg path {
    fill: ${themeColor('primary')};
  }
`

const MapDetailResultWrapper: FunctionComponent<MapDetailResultWrapperProps> = ({
  children,
  subTitle,
  title,
  onMaximize,
  location,
}) => {
  const { panoramaUrl, link, linkComponent } = useGetLegacyPanoramaPreview(location as any)
  const isEmbed = useSelector(isEmbedded)
  const user = useSelector(getUser)
  const { isForbidden, showComponent } = useCheckPanoramaPermission()

  return (
    <section className="map-detail-result">
      {showComponent && panoramaUrl && !isForbidden ? (
        <header
          className={`
      map-detail-result__header
      map-detail-result__header--${!isEmbed && panoramaUrl ? 'pano' : 'no-pano'}
    `}
        >
          <StyledLink
            to={link || ''}
            as={linkComponent}
            title={panoramaUrl ? 'Panoramabeeld tonen' : 'Geen Panoramabeeld beschikbaar'}
          >
            <img
              alt="Panoramabeeld"
              className="map-detail-result__header-pano"
              height="292"
              src={panoramaUrl}
              width="438"
            />
            <div className="map-detail-result__header-container">
              <h1 className="map-detail-result__header-title">{title}</h1>
              {subTitle && <h2 className="map-detail-result__header-subtitle">{subTitle}</h2>}
            </div>
          </StyledLink>
        </header>
      ) : (
        showComponent && isForbidden && <PanoAlert />
      )}
      <div className="map-detail-result__scroll-wrapper">
        {!user.authenticated && (
          <Header>
            <Heading styleAs="h4">{title}</Heading>
            {subTitle && (
              <Heading styleAs="h6" as="h2">
                {subTitle}
              </Heading>
            )}
          </Header>
        )}
        {children}
        <footer className="map-search-results__footer">
          <StyledButton type="button" onClick={onMaximize} iconLeft={<Maximize />} iconSize={21}>
            Volledig weergeven
          </StyledButton>
        </footer>
      </div>
    </section>
  )
}

export default MapDetailResultWrapper
