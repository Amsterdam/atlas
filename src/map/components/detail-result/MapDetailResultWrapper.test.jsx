import React from 'react'
import { shallow } from 'enzyme'

import MapDetailResultWrapper from './MapDetailResultWrapper'

jest.mock('redux-first-router-link', () => ({
  _esModule: true,
  default: jest.fn,
}))

jest.mock('react-redux', () => ({
  useSelector: jest.fn(() => []),
}))

describe('MapDetailResultWrapper', () => {
  it('should render everything', () => {
    const clickHandler = jest.fn()
    const wrapper = shallow(
      <MapDetailResultWrapper title="title" subTitle="subTitle" onMaximize={clickHandler}>
        <ul>
          <li>Item 1</li>
          <li>Item 2</li>
        </ul>
      </MapDetailResultWrapper>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should not render empty panoUrl', () => {
    const clickHandler = jest.fn()
    const wrapper = shallow(
      <MapDetailResultWrapper title="title" subTitle="subTitle" onMaximize={clickHandler}>
        <ul>
          <li>Item 1</li>
          <li>Item 2</li>
        </ul>
      </MapDetailResultWrapper>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should not render empty sub title', () => {
    const clickHandler = jest.fn()
    const wrapper = shallow(
      <MapDetailResultWrapper title="title" subTitle="" onMaximize={clickHandler}>
        <ul>
          <li>Item 1</li>
          <li>Item 2</li>
        </ul>
      </MapDetailResultWrapper>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should not render missing sub title', () => {
    const clickHandler = jest.fn()
    const wrapper = shallow(
      <MapDetailResultWrapper title="title" onMaximize={clickHandler}>
        <ul>
          <li>Item 1</li>
          <li>Item 2</li>
        </ul>
      </MapDetailResultWrapper>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should not render missing children', () => {
    const clickHandler = jest.fn()
    const wrapper = shallow(<MapDetailResultWrapper title="title" onMaximize={clickHandler} />)
    expect(wrapper).toMatchSnapshot()
  })
})
