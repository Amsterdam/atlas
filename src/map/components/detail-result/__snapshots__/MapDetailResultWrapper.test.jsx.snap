// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailResultWrapper should not render empty panoUrl 1`] = `
<section
  className="map-detail-result"
>
  <PanoAlert />
  <div
    className="map-detail-result__scroll-wrapper"
  >
    <styled.header>
      <Heading
        as="h1"
        styleAs="h4"
      >
        title
      </Heading>
      <Heading
        as="h2"
        styleAs="h6"
      >
        subTitle
      </Heading>
    </styled.header>
    <ul>
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
        type="button"
      >
        <span
          className="
                    map-search-results__button-icon
                    map-search-results__button-icon--maximize
                  "
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapDetailResultWrapper should not render empty sub title 1`] = `
<section
  className="map-detail-result"
>
  <PanoAlert />
  <div
    className="map-detail-result__scroll-wrapper"
  >
    <styled.header>
      <Heading
        as="h1"
        styleAs="h4"
      >
        title
      </Heading>
    </styled.header>
    <ul>
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
        type="button"
      >
        <span
          className="
                    map-search-results__button-icon
                    map-search-results__button-icon--maximize
                  "
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapDetailResultWrapper should not render missing children 1`] = `
<section
  className="map-detail-result"
>
  <PanoAlert />
  <div
    className="map-detail-result__scroll-wrapper"
  >
    <styled.header>
      <Heading
        as="h1"
        styleAs="h4"
      >
        title
      </Heading>
    </styled.header>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
        type="button"
      >
        <span
          className="
                    map-search-results__button-icon
                    map-search-results__button-icon--maximize
                  "
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapDetailResultWrapper should not render missing sub title 1`] = `
<section
  className="map-detail-result"
>
  <PanoAlert />
  <div
    className="map-detail-result__scroll-wrapper"
  >
    <styled.header>
      <Heading
        as="h1"
        styleAs="h4"
      >
        title
      </Heading>
    </styled.header>
    <ul>
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
        type="button"
      >
        <span
          className="
                    map-search-results__button-icon
                    map-search-results__button-icon--maximize
                  "
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapDetailResultWrapper should render everything 1`] = `
<section
  className="map-detail-result"
>
  <PanoAlert />
  <div
    className="map-detail-result__scroll-wrapper"
  >
    <styled.header>
      <Heading
        as="h1"
        styleAs="h4"
      >
        title
      </Heading>
      <Heading
        as="h2"
        styleAs="h6"
      >
        subTitle
      </Heading>
    </styled.header>
    <ul>
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
        type="button"
      >
        <span
          className="
                    map-search-results__button-icon
                    map-search-results__button-icon--maximize
                  "
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;
